apps/contrib/filters.py

```diff
@@ -11,76 +9,87 @@
 
 
 class StatusFilterWidget(widgets.DropdownLinkWidget):
-    label = _('Status')
+    label = _('Phase')
 
 
 class TopicFilterWidget(widgets.DropdownLinkWidget):
-    label = _('Topics')
+    label = _('Topic')
 
     def __init__(self, attrs=None):
-        choices = (('', _('Any')),)
+        choices = (('', _('All')),)
         choices += (models.abstracts.idea_section.
                     IDEA_TOPIC_CHOICES)
 
         super().__init__(attrs, choices)
 
 
-class YearFilterWidget(widgets.DropdownLinkWidget):
-    label = _('Year')
+class ProjectFilterWidget(widgets.DropdownLinkWidget):
+    label = _('Project')
 
-    def __init__(self, attrs=None):
-        choices = (('', _('Any')),)
-        now = datetime.now().year
-        years = []
-        try:
-            for project in Project.objects.all():
-                years += [project.created.year]
-        except Project.DoesNotExist:
-            years = [now]
-        for year in years:
-            choices += (year, year),
-        super().__init__(attrs, choices)
+
+class OrderingFilterWidget(widgets.DropdownLinkWidget):
+    label = _('Sorting')
+    right = True
 
 
 class IdeaFilterSet(DefaultsFilterSet):
 
-    defaults = {}
+    defaults = {
+        'ordering': 'newest'
+    }
 
     idea_topics = django_filters.CharFilter(
         lookup_expr='icontains',
         widget=TopicFilterWidget,
     )
 
-    project_year = django_filters.NumberFilter(
-        name='created',
-        lookup_expr='year',
-        widget=YearFilterWidget,
+    project = django_filters.ModelChoiceFilter(
+        name='module__project__name',
+        queryset=Project.objects.all(),
+        widget=ProjectFilterWidget,
     )
 
     def what_status(self, queryset, name, value):
-        if value == 'all':
-            qs = queryset.all()
-        elif value == 'idea_sketch':
-            qs = queryset.filter(proposal__isnull=True)
+        if value == 'idea_sketch':
+            qs = queryset.filter(is_proposal=False)
         elif value == 'proposal':
-            qs = queryset.filter(proposal__isnull=False)
+            qs = queryset.filter(is_proposal=True)
+        elif value == 'community_award':
+            qs = queryset.filter(community_award_winner=True)
+        elif value == 'camp':
+            qs = queryset.filter(visit_camp=True)
+        elif value == 'winner':
+            qs = queryset.filter(is_winner=True)
         else:
             qs = queryset.all()
```

Comment:
This case actually never happens since django's `ChoiceFilter`  seems to evaluate this in advance. If the value is not part of the choices, it returns an empty result, AFAIK.